### ISSUE TITLE
Allow JSON files as input for Secrets Stores and KVStores

### ISSUE THREAD

**Issue by GeeWee** [fastly/Viceroy#364]
Currently the `fastly.toml` [specification](https://www.fastly.com/documentation/reference/compute/fastly-toml/) allows you to pass in a JSON file for config stores (and earlier, also for dictionaries)

Something like that is extremely handy to be able to do, to e.g. share configuration across multiple fastly.tomls for environments, autogenerating these values from other scripts, not having to check them into CI etc.

Is that something you might be interested in adding back?

**Comment by fgsch** [comment]
This would be a great addition. Would you be interested in opening a PR for this?

**Comment by GeeWee** [comment]
I think I could try to find time for that if it would end up being accepted :)

**Comment by fgsch** [comment]
@GeeWee I'm pretty sure that'd be the case 👍

**Comment by GeeWee** [comment]
Okay, can you check? I'd hate to do the work and then have it languish :)

Do you have a proposal for the user-facing API in the TOML file? I'd like to lean against e.g. ConfigStores, but I think that's slightly harder, as that uses the format: "inline-toml" for keys defined inside the toml file.

**Comment by fgsch** [comment]
Sorry, looking at Viceroy I can see support for the `file` stanza for the secret store (https://github.com/fastly/Viceroy/pull/210) and kv store (originally added as `path` in https://github.com/fastly/Viceroy/pull/167 and later renamed for consistency in https://github.com/fastly/Viceroy/pull/206).
Is this what you wanted?

**Comment by GeeWee** [comment]
I don't _think_ so based at the documentation and the PR. I want to be able to specify a file with *all* my key-values, so e.g. a JSON object or similar, while the file property currently just allows me to specify the contents for a specific key - if I understand correctly?

**Comment by fgsch** [comment]
Ok, if you want all your key-values, that might require a bit of internal discussion as it is currently not part of the spec.
Sorry for the initial misunderstanding.

**Comment by GeeWee** [comment]
Any way I can push on that discussion, or if you have an idea about a timeline? This would bee extremely handy for us, and I was maybe a little too eager, so I already did the work to implement it in #365 😅

**Comment by kpfleming** [comment]
As I noted in the PR, we're just about to embark on a process to redesign the schema for fastly.toml to make this sort of thing easier to do (and make it more flexible as well), so I'd like to hold off a few weeks before we continue this discussion to see how this feature fits with that. Once we have an initial proposal we'll post it on our community forum (https://community.fastly.com) to get input from you and others who use Viceroy, and I'll update this ticket when that happens.

**Comment by joeshaw** [comment]
Related to (but distinct from) #359

**Comment by GeeWee** [comment]
@kpfleming do you have a rough timeline for this redesign?

**Comment by kpfleming** [comment]
There should be a post on our community forum in the next day or two, apologies for it taking much longer than planned to get the process started.

**Comment by GeeWee** [comment]
I'm unable to find the community forum post - any chance you have a link handy? :)

**Comment by kailan** [comment]
Hi @GeeWee –– apologies for the wait on this. I've just opened a community forum thread where we would welcome your input on `fastly.toml`: https://community.fastly.com/t/have-your-say-on-the-future-of-fastly-toml/2785

**Comment by GeeWee** [comment]
Thanks! Just gotten back from vacation - I've written my thoughts there. I assume there's no timeline for when this will change, and that the PR I've proposed won't be accepted until then?

**Comment by kpfleming** [comment]
We're aiming to close out the feedback/input process this week, and then draft a proposal for what the new format might look like... which will then lead to a comment/review period. Since that will take some time and your PR is relatively small, I'll work on getting it merged soon.

**Comment by kpfleming** [comment]
This is now partially complete, as the PR for JSON support for KV Stores has been merged, and the feature will be included in the next release.
